### PR TITLE
fix: More Fault Tolerance with OpenAI Schemas

### LINKED ISSUE #4327: [BUG] - Parsing with Ollama, "Unable to parse string as a number"
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

Wanted to utilize local ollama instance instead of openai, so:
 - created a pod
 - added an ollama container
    + made sure to pull the model I wanted (llama3.2:latest) to use and verify it does function
 - added mealie container with proper environmental variables
    + see that the mealie container does indeed spin up an ollama model, but throws errors

Attempted to run the parsing examples under the `/admin/parser/` page and get:

```
ERROR    2024-10-07T14:13:56 - Exception in ASGI application
Traceback (most recent call last):
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 24, in __call__
    await responder(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 44, in __call__
    await self.app(scope, receive, self.send_with_gzip)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/mealie/routes/parser/ingredient_parser.py", line 21, in parse_ingredient
    response = await parser.parse([ingredient.ingredient])
  File "/app/mealie/services/parser_services/openai/parser.py", line 99, in parse
    response = await self._parse(ingredients)
  File "/app/mealie/services/parser_services/openai/parser.py", line 84, in _parse
    responses = [
  File "/app/mealie/services/parser_services/openai/parser.py", line 85, in <listcomp>
    OpenAIIngredients.parse_openai_response(response_json) for response_json in responses_json if responses_json
  File "/app/mealie/schema/openai/_base.py", line 24, in parse_openai_response
    return cls.model_validate_json(response or "")
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/pydantic/main.py", line 597, in model_validate_json
    return cls.__pydantic_validator__.validate_json(json_data, strict=strict, context=context)
pydantic_core._pydantic_core.ValidationError: 1 validation error for OpenAIIngredients
ingredients.0.quantity
  Input should be a valid number, unable to parse string as a number [type=float_parsing, input_value='', input_type=str]
    For further information visit https://errors.pydantic.dev/2.8/v/float_parsing
```

### Mealie Version

podman `5.2.2` with mealie container `v1.12.0 `

### Deployment

Docker (Linux)

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

[Some LLMs](https://github.com/mealie-recipes/mealie/issues/4327) don't properly read our schema. While in general it should be the burden of the model/training-data to handle things correctly, we can be a bit more fault tolerant.

This PR coerces empty strings as a value of "0" for float fields.

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes https://github.com/mealie-recipes/mealie/issues/4327

### CODE REVIEW

**Comment by andrewdunndev** [comment]
I'll be very excited to provide feedback when this is released. Is there an incremental container build that I could test from this PR?

**Comment by michael-genson** [comment]
Yes, once this is merged you can try out the nightly branch. More info here: https://docs.mealie.io/documentation/getting-started/installation/installation-checklist/#docker-tags

File: mealie/schema/openai/recipe_ingredient.py [hunk 1]
```diff
@@ -72,8 +73,8 @@ class OpenAIIngredient(OpenAIBase):
         ),
     )
 
-    @field_validator("quantity")
-    def coerce_none_qty(cls, v: float | None) -> float:
+    @field_validator("confidence", "quantity", mode="before")
```
Comment (hay-kot) [inline mealie/schema/openai/recipe_ingredient.py:76]: <s>was this supposed to change `quantity` to `confidence`? Aren't there two confidence validators now? </s> I'm so stupid please ignore me

**Review by hay-kot — APPROVED** [review]

**Comment by andrewdunndev** [comment]
Can confirm this works with my local instance of ollama now. Sadly though it appears to be less valuable than using NLP or Brute. Is there a place that people are discussing the model performance with mealie parsing?

**Pull request merged**